src/liger_kernel/ops/cross_entropy.py

```diff
@@ -162,7 +173,12 @@ def liger_cross_entropy_kernel(
         X_block = tl.where(X_offsets != y, X_block, X_block - (1 - label_smoothing))
         # reduction scale
         if reduction == "mean":
-            X_block = X_block / (n_non_ignore)
+            if HAS_WEIGHT:
+                X_block = X_block / (sum_of_non_ignore_weight)
+            else:
+                X_block = X_block / (n_non_ignore)
+        if HAS_WEIGHT:
+            X_block = X_block * weight
```

Comment:
For my understanding is the multiplication of X_block with the weight here to help with the backward gradient computation?

Reply:
yes, it's part of the derivative of the unreduced loss [in the first equation](https://pytorch.org/docs/stable/generated/torch.nn.CrossEntropyLoss.html)